src/olympia/devhub/tests/test_tasks.py

```diff
@@ -76,34 +76,33 @@ def _uploader(resize_size, final_size):
     src = tempfile.NamedTemporaryFile(
         mode='r+w+b', suffix='.png', delete=False, dir=settings.TMP_PATH)
 
-    # resize_icon removes the original
-    shutil.copyfile(img, src.name)
-
-    src_image = Image.open(src.name)
-    assert src_image.size == original_size
-
-    if isinstance(final_size, list):
-        uploadto = user_media_path('addon_icons')
-        try:
-            os.makedirs(uploadto)
-        except OSError:
-            pass
-        for rsize, fsize in zip(resize_size, final_size):
-            dest_name = os.path.join(uploadto, '1234')
-
-            tasks.resize_icon(src.name, dest_name, resize_size, locally=True)
-            dest_image = Image.open(open('%s-%s.png' % (dest_name, rsize)))
-            assert dest_image.size == fsize
-
-            if os.path.exists(dest_image.filename):
-                os.remove(dest_image.filename)
-            assert not os.path.exists(dest_image.filename)
-        shutil.rmtree(uploadto)
-    else:
-        dest = tempfile.mktemp(suffix='.png', dir=settings.TMP_PATH)
-        tasks.resize_icon(src.name, dest, resize_size, locally=True)
-        dest_image = Image.open(dest)
-        assert dest_image.size == final_size
+    if not isinstance(final_size, list):
+        final_size = [final_size]
+        resize_size = [resize_size]
+    uploadto = user_media_path('addon_icons')
+    try:
+        os.makedirs(uploadto)
+    except OSError:
+        pass
+    for rsize, fsize in zip(resize_size, final_size):
```

Comment:
can we rename `fsize` `expected_size` instead since that's what we're testing here right? I'm a fan of the `assert given_value == expected_value` pattern and try to name things appropriately